consensus/types/src/beacon_state.rs

```diff
@@ -2580,11 +2581,12 @@ impl<E: EthSpec> BeaconState<E> {
         // for the internal nodes. Result should be 22 or 23, the field offset of the committee
         // in the `BeaconState`:
         // https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#beaconstate
-        let field_index = if self.fork_name_unchecked().electra_enabled() {
+        let field_gindex = if self.fork_name_unchecked().electra_enabled() {
             light_client_update::CURRENT_SYNC_COMMITTEE_INDEX_ELECTRA
         } else {
             light_client_update::CURRENT_SYNC_COMMITTEE_INDEX
         };
+        let field_index = field_gindex.safe_sub(self.num_fields_pow2())?;
```

Comment:
Should we make the GIndex a type to prevent these issues in the future?

Reply:
Yeah I think that sounds like a good idea. Maybe in a separate PR, as I'd like to keep this small and minimal for `v7.0.0-beta.1` which we should release ASAP

Reply:
For now I think the EF tests and index checks are good insurance against future mistakes